colossalai/pipeline/schedule/_utils.py

```diff
@@ -131,6 +131,16 @@ def retain_grad(x: Any) -> None:
         x.retain_grad()
 
 
+def require_grad(x: Any) -> None:
+    """Call require_grad on a tensor.
+
+    Args:
+        x (Any): Object to be called.
+    """
+    if isinstance(x, torch.Tensor) and x.requires_grad:
```

Comment:
`and not x.requires_grad`

Reply:
> `and not x.requires_grad`

Fixed in https://github.com/hpcaitech/ColossalAI/pull/6065/commits/6ee9584b9a2310bdd556ef32c9901828c2aec04d;